### ISSUE TITLE
Duration is cumulative when the learner retries a question

### ISSUE THREAD

**Issue by garemoko** [h5p/h5p-php-library#5]
Some content types allow the user to re-attempt a question and send a statement after each one. Here, the duration is calculated as the time between when the learner accessed the question first and when they submitted the attempt, rather than between when they started and submitted an attempt. This will cause inaccuracies for systems reporting on the data that add durations to calculate total time. 

I suggest adding something [here](https://github.com/h5p/h5p-php-library/blob/master/js/h5p-x-api-event.js#L33) to either reset instance.activityStartTime or store another variable such as attemptStartTime. Presumably that's be a new property of instance? Can we modify instance in this context?

**Comment by falcon-git** [comment]
Good point. I think we will have to change all question types that supports retry to set attemptStartTime on retry. Hopefully this can be done for many of the question types from the "super class" H5P.Question, but I'm not sure... If we do this we probably should trigger some kind of retry event as well, but I'm not sure which verb to use...

**Comment by garemoko** [comment]
Most/all content I've seen uses the same verb for retries. I don't think it would hurt to use a different one though.

**Comment by falcon-git** [comment]
I'm not sure if we're on the same page. I meant when the retry button is clicked. I guess "attempted" would be the right statement to trigger, and we probably should trigger that the first time the user starts on a task as well...

**Comment by falcon-git** [comment]
By the way our sponsor agrees with your changes so we'll be merging them in as soon as we've tested them.

**Comment by garemoko** [comment]
Yes, I had misunderstood. Got it now. 'Attempted' sounds good!
